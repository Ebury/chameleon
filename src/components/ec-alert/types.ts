export enum PropsAlertType {
  TEXT = 'text',
  DATE = 'date',
  NUMBER = 'number',
  TEL = 'tel'
}

export enum AlertType {
    ERROR = 'error', 
    INFO = 'info', 
    SUCCESS = 'success', 
    WARNING = 'warning'
}

export enum AlertEvent {
  UPDATE_OPEN = 'update:open',
  ACTION = 'action',
  CHANGE = 'change',
}

export interface AlertEvents {
  [AlertEvent.UPDATE_OPEN]: AlertProps['open']
  [AlertEvent.ACTION]: undefined
  [AlertEvent.CHANGE]: AlertProps['open']
}

export interface AlertProps {
  propsType?: PropsAlertType,
  type: AlertType,
  title: string,
  subtitle?: string,
  dismissable?: boolean,
  buttonText?: string,
  open?: boolean,
  responsive?: boolean
}
