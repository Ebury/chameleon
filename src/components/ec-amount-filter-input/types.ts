import type { DropdownItem } from '../ec-dropdown/types';

export interface ComparisonSymbolItem extends DropdownItem<string> {}

export interface AmountFilterInputModel {
  amount?: number,
  comparisonSymbol?: ComparisonSymbolItem,
}

export interface AmountFilterInputProps {
  modelValue: AmountFilterInputModel,
  locale?: string,
  isSensitive?: boolean,
  label?: string,
  note?: string,
  amountPlaceholder?: string,
  bottomNote?: string,
  isWarning?: boolean,
  warningTooltipMessage?: string,
  errorMessage?: string,
  errorTooltipMessage?: string,
  comparisonSymbolItems: ComparisonSymbolItem[],
}
