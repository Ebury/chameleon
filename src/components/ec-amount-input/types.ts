import type { InputFieldProps } from '../ec-input-field/types';

export interface AmountInputProps extends Omit<InputFieldProps, 'modelValue'> {
  modelValue?: number | string | null,
  isMasked?: boolean,
  locale?: string,
  currency?: string,
}
