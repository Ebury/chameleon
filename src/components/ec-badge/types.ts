import { AlertType as BadgeType } from '../ec-alert/types';

export { BadgeType };

export interface BadgeProps {
  type?: BadgeType,
  value: string | number,
}
