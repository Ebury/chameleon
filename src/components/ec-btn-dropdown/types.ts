import type { AnchorHTMLAttributes } from 'vue';
import type { RouteLocationRaw } from 'vue-router';

import type { DropdownItem } from '../ec-dropdown-search/types';

export interface BtnDropdownItem<TValue = string> extends DropdownItem<TValue> {
  href?: string,
  to?: string | RouteLocationRaw,
  attrs?: AnchorHTMLAttributes,
}

export interface BtnDropdownProps<TValue = string, TBtnDropdownItem extends BtnDropdownItem<TValue> = BtnDropdownItem<TValue>> {
  items?: TBtnDropdownItem[],
  isBtnDropdownDisabled?: boolean,
  buttonText: string,
  listDataTest?: string,
  href?: string,
  to?: string | RouteLocationRaw,
}
