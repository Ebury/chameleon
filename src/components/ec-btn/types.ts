import type { RouteLocationRaw } from 'vue-router';

import type { IconName } from '../ec-icon/iconNames';

export enum ButtonSize {
  Medium = 'md',
  Small = 'sm'
}

export enum ButtonCategory {
  Primary = 'primary',
  Secondary = 'secondary',
  Success = 'success',
  Error = 'error',
  Warning = 'warning'
}

export interface ButtonProps {
  size?: ButtonSize;
  href?: string;
  to?: RouteLocationRaw;
  tag?: string;
  isDisabled?: boolean;
  icon?: IconName;
  isRounded?: boolean;
  isOutline?: boolean;
  isFullWidth?: boolean;
  category: ButtonCategory;
  isReverse?: boolean;
  isLoading?: boolean;
  isSubmit?: boolean
}
