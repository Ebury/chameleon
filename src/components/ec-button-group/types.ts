export interface ButtonGroupItem<TValue> {
  id?: string | number,
  value: TValue,
  text: string,
  disabled?: boolean,
}

export interface ButtonGroupProps<TValue> {
  modelValue?: TValue,
  items: ButtonGroupItem<TValue>[],
}
