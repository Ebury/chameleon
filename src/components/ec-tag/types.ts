import type { TooltipOptions } from '../../directives/ec-tooltip/types';
import type { IconName, IconType } from '../ec-icon/types';

export interface TagProps {
  text: string,
  isIconRounded?: boolean,
  iconName?: IconName,
  iconType?: IconType,
  tooltipOptions?: TooltipOptions,
}
