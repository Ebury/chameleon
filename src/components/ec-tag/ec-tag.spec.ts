import { mount } from '@vue/test-utils';

import { EcTooltipDirectiveMock } from '../../../tests/mocks/ec-tooltip.mock';
import type { CVueWrapper } from '../../../tests/utils/global';
import { IconName } from '../ec-icon/icon-names';
import { IconType } from '../ec-icon/types';
import EcTag from './ec-tag.vue';
import type { TagProps } from './types';

describe('EcTag', () => {
  function mountTag(props?: TagProps) {
    return mount(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      EcTag as any,
      {
        props,
        global: {
          mocks: {
            vEcTooltip: EcTooltipDirectiveMock,
          },
        },
      },
    ) as CVueWrapper;
  }

  describe(':props', () => {
    it('should render as expected when passing the text prop', () => {
      const wrapper = mountTag({
        text: 'Trusted',
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render as expected when passing the icon name', () => {
      const wrapper = mountTag({
        text: 'Trusted',
        iconName: IconName.RoundedCheck,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render as expected when passing the icon type', () => {
      const wrapper = mountTag({
        text: 'Trusted',
        iconName: IconName.RoundedCheck,
        iconType: IconType.SUCCESS,
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render as expected when we set "is-icon-rounded" to be true', () => {
      const wrapper = mountTag({
        text: 'Trusted',
        iconName: IconName.CurrencyAed,
        isIconRounded: true,
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('should render with a tooltip', () => {
    const text = 'Trusted';

    const wrapper = mountTag({
      text,
    });

    expect(wrapper.findByDataTest('ec-tag').attributes('data-ec-tooltip-mock-content')).toBe(text);
    expect(wrapper.element).toMatchSnapshot();
  });
});
