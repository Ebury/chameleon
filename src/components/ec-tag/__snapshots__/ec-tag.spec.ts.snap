// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcTag > :props > should render as expected when passing the icon name 1`] = `
<div
  class="ec-tag"
  data-ec-tooltip-mock-content="Trusted"
  data-test="ec-tag ec-mock ec-tooltip-mock"
>
  <svg
    class="ec-icon ec-tag__icon"
    data-test="ec-tag__icon"
    height="16"
    width="16"
  >
    <use
      xlink:href="#ec-rounded-check"
    />
  </svg>
  <div
    class="ec-tag__text"
    data-test="ec-tag__text"
  >
    Trusted
  </div>
</div>
`;

exports[`EcTag > :props > should render as expected when passing the icon type 1`] = `
<div
  class="ec-tag"
  data-ec-tooltip-mock-content="Trusted"
  data-test="ec-tag ec-mock ec-tooltip-mock"
>
  <svg
    class="ec-icon ec-icon--success ec-tag__icon"
    data-test="ec-tag__icon"
    height="16"
    width="16"
  >
    <use
      xlink:href="#ec-rounded-check"
    />
  </svg>
  <div
    class="ec-tag__text"
    data-test="ec-tag__text"
  >
    Trusted
  </div>
</div>
`;

exports[`EcTag > :props > should render as expected when passing the text prop 1`] = `
<div
  class="ec-tag"
  data-ec-tooltip-mock-content="Trusted"
  data-test="ec-tag ec-mock ec-tooltip-mock"
>
  <!--v-if-->
  <div
    class="ec-tag__text"
    data-test="ec-tag__text"
  >
    Trusted
  </div>
</div>
`;

exports[`EcTag > :props > should render as expected when we set "is-icon-rounded" to be true 1`] = `
<div
  class="ec-tag"
  data-ec-tooltip-mock-content="Trusted"
  data-test="ec-tag ec-mock ec-tooltip-mock"
>
  <svg
    class="ec-icon ec-tag__icon tw-rounded-1/2"
    data-test="ec-tag__icon"
    height="16"
    width="16"
  >
    <use
      xlink:href="#ec-currency-aed"
    />
  </svg>
  <div
    class="ec-tag__text"
    data-test="ec-tag__text"
  >
    Trusted
  </div>
</div>
`;

exports[`EcTag > should render with a tooltip and use text as content 1`] = `
<div
  class="ec-tag"
  data-ec-tooltip-mock-content="Trusted"
  data-test="ec-tag ec-mock ec-tooltip-mock"
>
  <!--v-if-->
  <div
    class="ec-tag__text"
    data-test="ec-tag__text"
  >
    Trusted
  </div>
</div>
`;

exports[`EcTag > should render with a tooltip that has a different content 1`] = `
<div
  class="ec-tag"
  data-ec-tooltip-mock-content="Tooltip text"
  data-test="ec-tag ec-mock ec-tooltip-mock"
>
  <!--v-if-->
  <div
    class="ec-tag__text"
    data-test="ec-tag__text"
  >
    Trusted
  </div>
</div>
`;
